File: extra/Problems /Problem 2/array_Insert.py
'''
# Sample code to perform I/O:
 
name = raw_input()          # Reading input from STDIN
print 'Hi, %s.' % name      # Writing output to STDOUT
 
# Warning: Printing unwanted or ill-formatted data to output will cause the test cases to fail
'''
 
# Write your code here
num_queries = int(raw_input().split(' ')[1])
my_array = map(int, raw_input().split(' '))
for i in range(num_queries):
    query = map(int, raw_input().split(' '))
    if query[0] == 1:
        index = query[1]
        new_num = query[2]
        my_array[index] = new_num
    if query[0] == 2:
        sum = 0
        for j in range(query[1], query[2]+1):
            sum += my_array[j]
        print sum

File: SAProject/Problems /Problems /Problem 2/array_Insert.pl
=comment
# Sample code to perform I/O:
 
my $name = <STDIN>;             # Reading input from STDIN
print "Hi, $name.\n";           # Writing output to STDOUT
 
# Warning: Printing unwanted or ill-formatted data to output will cause the test cases to fail
=cut
 
# Write your code here
use strict;
use warnings;
 
my ($arr_length, $query_num) = split(/\s/, <>);
my @arr = split(/\s/, <>);
my $count = 0;
for(1..$query_num) {
    my ($query_type, $arg1, $arg2) = split(/\s/, <>);
    if ($query_type == 1) {
        $arr[$arg1] = $arg2;
        $count++;
    } elsif ($query_type == 2) {
        my $sum = 0;
        for($arg1..$arg2) {
            $sum += $arr[$_];
        }
        $count++;
        print $sum . "\n";
    }
}